.github/workflows/mypy-clinic.yml

```
@@ -0,0 +1,37 @@
+name: Run mypy on Tools/clinic
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - "Tools/clinic/**/*"
```

Or this?

```suggestion
      - "Tools/clinic/**"
```

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet